html/modules/custom/reliefweb_import/src/Service/WorkDayJobImporter.php

```diff
@@ -0,0 +1,326 @@
+        ];
+      }
+
+      return $documents;
+    }
+    catch (\Exception $e) {
+      $this->getLogger()->error('Unable to retrieve the WorkDay documents: ' . $e->getMessage());
+      return [];
+    }
+  }
+
+  /**
+   * Process the WorkDay documents data.
+   *
+   * @param array $documents
+   *   The WorkDay documents data.
+   */
+  public function importDocuments(array $documents): void {
+    $source_id = $this->settings['source_id'] ?? 0;
+    $uid = $this->settings['uid'] ?? 2;
+
+    $errors = [];
+    $warnings = [];
+
+    foreach ($documents as $item) {
+      $name = $item['title'] ?? 'unknown';
+      try {
+        $guid = trim($item['url'] ?? '');
+        $this->url = $guid;
+
+        // Check if job already exist.
+        if ($this->jobExists($guid)) {
+          $this->getLogger()->notice(strtr('Updating job @guid', [
+            '@guid' => $guid,
+          ]));
+          $job = $this->loadJobByGuid($guid);
+          if (empty($job)) {
+            throw new ReliefwebImportExceptionViolation(strtr('Unable to load job @guid', [
+              '@guid' => $guid,
+            ]));
+          }
+          $this->updateJob($job, $item);
+        }
+        else {
+          $this->getLogger()->notice(strtr('Creating new job @guid', [
+            '@guid' => $guid,
+          ]));
+          $this->createJob($guid, (object) $item, $uid, $source_id);
+        }
+      }
+      catch (ReliefwebImportExceptionViolation $exception) {
+        $errors[] = $exception->getMessage();
+      }
+      catch (ReliefwebImportExceptionSoftViolation $exception) {
+        $warnings[] = $exception->getMessage();
+      }
+    }
+
+    if (!empty($errors)) {
+      $this->getLogger()->error(strtr('Errors while processing @name: @errors', [
+        '@name' => $name,
+        '@errors' => "\n- " . implode("\n- ", $errors),
+      ]));
+    }
+    if (!empty($warnings)) {
+      $this->getLogger()->warning(strtr('Warnings while processing @name: @warnings', [
+        '@name' => $name,
+        '@warnings' => "\n- " . implode("\n- ", $warnings),
+      ]));
+    }
+  }
+
+  /**
+   * Create a new job.
+   *
+   * @param string $guid
+   *   Feed item unique ID.
+   * @param object|array $data
+   *   Data for the job.
+   * @param int $uid
+   *   ID of the job owner.
+   * @param int $source_id
+   *   Source ID.
+   */
+  protected function createJob(string $guid, object|array $data, int $uid, int $source_id): void {
+    $values = [
+      'type' => 'job',
+      'uid' => $uid,
+      'field_source' => $source_id,
+      'field_import_guid' => $guid,
+    ];
+    $job = $this->entityTypeManager->getStorage('node')->create($values);
+    $this->updateJob($job, $data);
+  }
+
+  /**
+   * Update an existing job.
+   *
+   * @param \Drupal\reliefweb_entities\Entity\Job $job
+   *   Job to update.
+   * @param object|array $data
+   *   Data for the job.
+   */
+  protected function updateJob(Job $job, object|array $data): void {
```

Comment:
The data type doesn't seem correct. The code lines 302-304 expect an object and the function doesn't have any casting from array to object.